packages/valory/skills/abstract_round_abci/models.py

```diff
@@ -280,29 +540,20 @@ class TendermintRecoveryParams:
     reset_params: Optional[List[Tuple[str, str]]] = None
 
 
-class ApiSpecs(Model):
+class ApiSpecs(Model, FrozenMixin, TypeCheckMixin):
     """A model that wraps APIs to get cryptocurrency prices."""
 
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         """Initialize ApiSpecsModel."""
-        self.url: str = self.ensure("url", kwargs)
-        self.api_id: str = self.ensure("api_id", kwargs)
-        self.method: str = self.ensure("method", kwargs)
-        self.headers: HeadersType = kwargs.pop("headers", [])
-        self.parameters: ParametersType = kwargs.pop("parameters", [])
-        self.response_info = ResponseInfo(kwargs)
-        self.retries_info = RetriesInfo(kwargs)
-
+        self.url: str = self._ensure("url", kwargs, str)
+        self.api_id: str = self._ensure("api_id", kwargs, str)
+        self.method: str = self._ensure("method", kwargs, str)
+        self.headers: HeadersType = self._ensure("headers", kwargs, list)
+        self.parameters: ParametersType = self._ensure("parameters", kwargs, list)
```

Comment:
Todo: check actual types

Reply:
Are we doing this here, or are we creating an issue for it?

Reply:
Will do in separate PR